lib/http-bridge/test/integration/Test/Integration/Scenario/CLISpec.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Integration.Scenario.CLISpec
+    ( specNoCluster
+    , specWithCluster
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Types
+    ( ApiWallet )
+import Control.Monad
+    ( forM_ )
+import Data.List
+    ( length )
+import Data.List.Split
+    ( splitOn )
+import Data.Text
+    ( Text )
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe, shouldContain, shouldNotContain )
+import System.Command
+    ( Stdout (..), command )
+import Test.Hspec
+    ( SpecWith, describe, it )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , expectResponseCode
+    , getFromResponse
+    , json
+    , request
+    , walletId
+    )
+
+import qualified Data.Text as T
+import qualified Network.HTTP.Types.Status as HTTP
+
+specNoCluster :: SpecWith ()
+specNoCluster = do
+
+    it "CLI - Shows version" $  do
+        Stdout out <- command [] "cardano-wallet" ["--version"]
+        cabal <- readFile "../../cardano-wallet.cabal"
+        cabal `shouldContain` out
+
+    it "CLI - Can generate mnemonics with default size" $  do
+        Stdout out <- command [] "cardano-wallet" ["mnemonic", "generate"]
+        length (splitOn " " out) `shouldBe` 15
+
+    describe "CLI - Can generate mnemonics with different sizes" $  do
+        forM_ [9, 12, 15, 18, 21, 24] $ \size -> it (show size) $ do
+            Stdout out <- command [] "cardano-wallet"
+                            ["mnemonic", "generate", "--size", show size]
+            length (splitOn " " out) `shouldBe` size
```

Comment:
Would deserve some negative tests too I think (not saying this should be part of _this_ PR, but for exhaustiveness)